util/misc.py

```diff
@@ -410,4 +412,106 @@ def which(application_binary_name):
         full_path=os.path.join(path, application_binary_name)
         if os.path.exists(full_path) and os.access(full_path, os.X_OK):
             link_resolved_path = os.path.realpath(full_path)
-            return link_resolved_path
\ No newline at end of file
+            return link_resolved_path
+
+def is_nonstr_seq(x):
+    '''Tests whether `x` is a Sequence other than a string'''
+    return isinstance(x, collections.Sequence) and not isinstance(x,(str,bytes))
+
+def make_seq(x):
+    '''Return a tuple containing the items in `x`, or containig just `x` if `x` is a non-string sequence.  Convenient
+    for uniformly writing iterations over parameters that may be passed in as an item or a tuple/list of items.
+    '''
+    return tuple(x) if is_nonstr_seq(x) else (x,)
+
+def flatten_dict(d):
+    '''Return a new dict `r`, where r[(k1,k2,...,kn)]==v iff d[k1][k2]...[kn]==v.'''
+    result=dict()
+    for k, v in d.items():
+        assert not isinstance(k,tuple), 'Dict already flattened? {}'.format(d)
```

Comment:
You can in general.  The utila here are just subroutines for handling config dicts, which always have string keys, while flattened dicts always have tuple keys; this can be used as a sanity check.  I'move it into a unit test.